libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -72,7 +72,31 @@ class ByteBuffer {
       read len bytes without advancing the read pointer
     */
     uint32_t peekbytes(uint8_t *data, uint32_t len);
-    
+
+    // Similar to peekbytes(), but will fill out IoVec struct with
+    // both parts of the ring buffer if wraparound is happpening, or
+    // just one part. Returns the number of parts written to.
+    struct IoVec {
+        uint8_t *data;
+        uint32_t len;
+    };
+    int peekiovec(IoVec vec[2], uint32_t len);
```

Comment:
Ohh, the other method is peekbytes? Ok, they may be similar but the meaning of each return value is totally orthogonal. One returns the size that were successfully "peeked" while the other returns the number of vectors from IoVec that were filled with "peeked" bytes. Force it's return type/size to be the same can actually lead to silly mistakes, it may lead someone to think that peekiovec also returns a length.